serde-api/src/main/java/io/micronaut/serde/Serializer.java

```diff
@@ -120,5 +122,23 @@ default boolean hasView(Class<?>... views) {
         <B, P> SerializationReference<B, P> resolveReference(
                 @NonNull SerializationReference<B, P> reference
         );
+
+        /**
+         * Get the {@link SerdeConfiguration} for this context.
+         *
+         * @return The {@link SerdeConfiguration}
+         * @since 2.6.0
+         */
+        @NonNull
+        SerdeConfiguration getSerdeConfiguration();
+
+        /**
+         * Get the {@link SerializationConfiguration} for this context.
+         *
+         * @return The {@link SerializationConfiguration}
+         * @since 2.6.0
+         */
+        @NonNull
+        SerializationConfiguration getSerializationConfiguration();
```

Comment:
Looking at the the use-cases, I think it might be better to add those configurations to `createSpecific` method instead to the context

Reply:
i dont understand what you mean, i dont want to add more arguments to createSpecific.

Reply:
IMHO it would be better to have it in `createSpecific` than in the context, as it's not really contextual

Reply:
createSpecific has so many call sites, basically any serde that uses another serde calls it. They would all have to be amended with the new config arguments.

The whole point of the EncoderContext imo is to encapsulate information that is broader in scope than a single Encoder but more narrow than the BeanContext. that now fits these config classes.